chain/sync.go

```diff
@@ -1618,8 +1618,8 @@ func (syncer *Syncer) fetchMessages(ctx context.Context, headers []*types.TipSet
 				}
 
 				mx.Lock()
-				if requestResult != nil {
-					copy(batch[j+offset:], requestResult)
+				if requestResult != nil && nextI+len(requestResult) <= lastI {
```

Comment:
I don't think this will work with partial responses (which we want to allow)

Reply:
this just breaks things.